src/main/java/org/ibp/api/java/impl/middleware/ontology/validator/VariableValidator.java

```diff
@@ -152,6 +162,61 @@ private boolean nameValidationProcessor(final VariableDetails variable, final Er
 		return errors.getErrorCount() == initialCount;
 	}
 
+	private boolean aliasValidationProcessor(final VariableDetails variable, final Errors errors) {
+
+		if (StringUtils.isBlank(variable.getAlias())) {
+			return true;
+		}
+
+		if(variable.getAlias().equalsIgnoreCase(variable.getName())){
+			this.addCustomError(errors, "alias", BaseValidator.NAME_AND_ALIAS_ARE_EQUALS, null);
+		}
+
+		final Integer initialCount = errors.getErrorCount();
+
+		// Trim name
+		variable.setAlias(variable.getAlias().trim());
+
+		// Name is no more than 32 characters
+		this.fieldShouldNotOverflow("alias", variable.getAlias(), VariableValidator.NAME_TEXT_LIMIT, errors);
+
+		if (errors.getErrorCount() > initialCount) {
+			return false;
+		}
+
+		// Name must only contain alphanumeric characters, underscores and cannot start with a number
+		this.fieldShouldHaveValidPattern("alias", variable.getAlias(), "alias", errors);
+
+		if (errors.getErrorCount() > initialCount) {
+			return false;
+		}
+
+		// The name must be unique
+		this.checkTermUniqueness("Alias", "a Name", StringUtil.parseInt(variable.getId(), null), variable.getAlias(),
+			CvId.VARIABLES.getId(), errors); //
+
+		if (errors.getErrorCount() > initialCount) {
+			return false;
+		}
+
+		// The alias must be unique
+		this.checkVariableAliasUniqueness("Alias", "an Alias", variable.getId(), variable.getAlias(), variable.getProgramUuid(), errors);
+
+		return errors.getErrorCount() == initialCount;
+	}
+
+	private void checkVariableAliasUniqueness(final String fieldName, final String termName, final String variableId, final String alias, final String programUUUid, final Errors errors) {
+		final Integer varId = StringUtils.isNotBlank(variableId) ? Integer.valueOf(variableId) : null;
+		final List<VariableOverridesDto> variableOverridesList =
```

Comment:
if varId is not an instance of Integer, then there is no need to run any query
doing this allows to simplify the IF condition

Reply:
done.